docs/libcudacxx/extended_api/synchronization_primitives/barrier.rst

```diff
@@ -160,4 +160,136 @@ For each :ref:`cuda::thread_scope <libcudacxx-extended-api-memory-model-thread-s
      cuda::barrier<cuda::thread_scope_block> d(10);
    }
 
-`See it on Godbolt <https://godbolt.org/z/ehdrY8Kae>`_
+`See it on Godbolt <https://godbolt.org/z/7Kbz5qqhh>`_
+
+.. rubric:: Example: 1D TMA load of two buffers with arrival token (sm90+)
+
+The following example shows how to use TMA to load two tiles of data from global memory into shared memory:
+
+.. code:: cuda
+
+   #include <cuda/barrier>
+   #include <cuda/ptx>
+
+   // selects a single leader thread from the block
+   __device__ bool elect_one() {
+     // elect_sync is important to help the optimizer generate a uniform datapath
+     return cuda::ptx::elect_sync(~0) && threadIdx.x < 32;
+   }
+
+   __global__ void example_kernel(int* gmem1, double* gmem2) {
+     constexpr int tile_size = 1024;
+     __shared__ alignas(16)    int smem1[tile_size];
+     __shared__ alignas(16) double smem2[tile_size];
+
+     #pragma nv_diag_suppress static_var_with_dynamic_init
+     using barrier_t = cuda::barrier<cuda::thread_scope_block>;
+     __shared__  barrier_t bar;
+
+     // setup the barrier where each thread in the block arrives at
+     if (elect_one()) {
```

Comment:
We can cache `elect_one()` for reuse later:

```
bool is_block_leader = elect_one()
[...]
if (is_warp_leader) {
[...]
)
[...]
```

Reply:
Addressed in #6329